src/Security/Helper/FinishRegistrationBehavior.php

```diff
@@ -31,10 +32,13 @@ trait FinishRegistrationBehavior
     protected function saveUserInfoToSession(Request $request, FinishRegistrationException $e)
     {
         // save the user information!
-        $request->getSession()->set(
-            'guard.finish_registration.user_information',
-            $e->getUserInformation()
-        );
+        $session = $request->getSession();
+        if ($session instanceof SessionInterface) {
+            $session->set(
+                'guard.finish_registration.user_information',
+                $e->getUserInformation()
+            );
+        }
```

Comment:
What if not? Probably better to throw an exception else? And the same in a few other places. Wdyt?

Reply:
indeed - OAuth2Client::getSession() will throw a \LogicException if there is no session - and it's 'stateless''. Should a session be required everywhere - and how much stateless-ness is needed?

Reply:
Hmm, in this case, because the user would be calling this method directly (and the method is called `saveUserInfoToSession()`), if we're unable to actually save to the session because there is none, I think an exception is a good idea

Reply:
Taking a look in the `Request` class, Symfony 4.1 deprecated calling `getSession()` if there one didn't exist - it's been triggering a notice since then, and 5.0 will throw a `BadMethodCallException`.  Till 4.4 or so though, it was  however, still marked as potentially returning null.

I'll make a friendlier error message instead for all the checks with a `if (!$request->hasSession()) { throw .... }`.  The`hasSession()` method was in for our minimum-supported version of 3.4.